mobile/lib/models/integrations/discord.integration.model.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:triggo/models/integration.model.dart';
+import 'package:triggo/repositories/models/integration.repository.model.dart';
+import 'package:triggo/repositories/models/integrations/discord.integrations.dart';
+
+class DiscordIntegration extends Integration {
+  final String username;
+  final String email;
+  final String displayName;
+  final String avatarUri;
+  final List<String> flags;
+
+  DiscordIntegration({
+    required super.name,
+    required this.username,
+    required this.email,
+    required this.displayName,
+    required this.avatarUri,
+    required this.flags,
+  });
+
+  static DiscordIntegration fromDTO(IntegrationDTO dto) {
+    DiscordProperties properties = dto.properties as DiscordProperties;
+    return DiscordIntegration(
+      name: 'Discord',
```

Comment:
I remember you making a class or enum or listing all the integrations. Can we use this directly? I'm not sure.

Reply:
You might think about this method:
```dart
factory IntegrationDTO.fromJson(Map<String, dynamic> json) {
    IntegrationPropertiesDTO properties;
    switch (json['type']) {
      case IntegrationType.discord:
        properties = DiscordProperties.fromJson(json['properties']);
        break;
      default:
        throw Exception('Unknown integration type');
    }
}
```

If not, I'm finding what you're telling me

Reply:
Mmmhh kinda. But never mind, It's great